dash/resources.py

```diff
@@ -14,48 +14,73 @@ def __init__(self, resource_name):
     def append_resource(self, resource):
         self._resources.append(resource)
 
+    # pylint: disable=too-many-branches
     def _filter_resources(self, all_resources, dev_bundles=False):
         filtered_resources = []
         for s in all_resources:
             filtered_resource = {}
             if 'dynamic' in s:
                 filtered_resource['dynamic'] = s['dynamic']
+            if 'async' in s:
+                if 'dynamic' in s:
+                    raise exceptions.ResourceException(
+                        '{} can\'t have both \'dynamic\' and '
+                        '\'async\'.'.format(json.dumps(filtered_resource))
```

Comment:
async and dynamic are incompatible

Reply:
🐄 moot point if we `black` this, but if you find yourself using `\'`, switch the outer quotes to `"`

Also the error msg would probably be easier to read with the json dump at the end so you don't have to hunt for the error.